lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -1048,16 +1050,13 @@ def get(self, request, subsection_id):
         try:
             original_grade = PersistentSubsectionGrade.read_grade(user_id, usage_key)
         except PersistentSubsectionGrade.DoesNotExist:
-            results = SubsectionGradeResponseSerializer({
-                'original_grade': None,
-                'override': None,
-                'history': [],
-                'subsection_id': usage_key,
-                'user_id': user_id,
-                'course_id': None,
-            })
-
-            return Response(results.data)
+            student = get_user_model().objects.get(id=user_id) 
+            from lms.djangoapps.course_blocks.api import get_course_blocks
+            course_structure = get_course_blocks(student, usage_key)
+            course = modulestore().get_course(usage_key.course_key)
+            subsection_grade_factory = SubsectionGradeFactory(student, course=course, course_structure=course_structure)
+            grade = subsection_grade_factory.create(course_structure[usage_key], force_caclulate=True)
```

Comment:
1. `force_caclulate` is a typo, should be `force_calculate` :) . But having a `force_calculate` flag is generally the right idea - we need to force past that `assume_zero_if_absent()` check in the `create()` method.
2. we should be calling `subsection_grade_factory.create()` with `read_only=True` - we don't want to create a `PersistentSubsectionGrade` record as part of this request.
3. after calling `create()` here, on a (student, subsection) who hasn't yet attempted the subsection, you should have a `CreateSubsectionGrade` object stored in the `grade` variable that has the possible grade information inside it's `graded_total` attribute:
```(Pdb) grade.graded_total
AggregatedScore({'first_attempted': None, 'graded': True, 'possible': 6.0, 'earned': 0.0})```
(you should try this out locally using PDB, it's informative)
4. At this point, we have to do something different.  Remember, we shouldn't be creating a `PersistentSubsectionGrade` here - that's for another endpoint to handle.  So there won't be any model in `grade`.  Instead, we have to change what's passed into the `SubsectionGradeResponseSerializer` - right now it expects to only deal with the grade models, and not any object that inherits from `SubsectionGrade`.  We should probably change that. It looks like the serializer ends up giving us a dict of earned/possible graded/all points in the `original_grade` key (see the docstring of this class), so maybe we can just pass in a dictionary, instead of a `PersistentSubsectionGrade` object.  That way, when the line above `PersistentSubsectionGrade.read_grade(user_id, usage_key)` returns something, you can just pluck the scores out of it and create a dict, which you pass to the serializer, and in the case where there is no attempt, you can pluck the scores from the `CreateSubsectionGrade` `all_total` and `graded_total` attributes.

I know that's a lot, let me know if you have any questions.